bin/helpers/utils.js

```diff
@@ -311,25 +325,187 @@ exports.getLocalFlag = (connectionSettings) => {
+      var bs_local_args = this.setLocalArgs(bsConfig, args);
+      let that = this;
+      logger.info('Setting up Local testing...');
+      bs_local.start(bs_local_args, function (localStartError) {
+        if (that.isUndefined(localStartError)) {
+          resolve(bs_local);
+        } else {
+          let message = `name: ${localStartError.name}, message: ${localStartError.message}, extra: ${localStartError.extra}`,
+              errorCode = "local_start_error";
+          that.sendUsageReport(
+            bsConfig,
+            args,
+            message,
+            Constants.messageTypes.ERROR,
+            errorCode
+          );
+          reject(localStartError);
+        }
+      });
+    } else {
+      resolve();
+    }
+  });
+};
+
+exports.stopLocalBinary = (bsConfig, bs_local) => {
+  return new Promise((resolve, reject) => {
+    if (!this.isUndefined(bs_local) && bs_local.isRunning() && bsConfig['connection_settings']['local_mode'].toLowerCase() != "always-on") {
+      let that = this;
+      bs_local.stop(function (localStopError) {
+        if (that.isUndefined(localStopError)) {
+          resolve();
+        } else {
+          let message = `name: ${localStartError.name}, message: ${localStartError.message}, extra: ${localStartError.extra}`,
+              errorCode = "local_stop_error";
+          that.sendUsageReport(
+            bsConfig,
+            args,
+            message,
+            Constants.messageTypes.ERROR,
+            errorCode
+          );
+        }
+      });
+    } else {
+      resolve();
+    }
+  });
+};
+
+exports.setLocalArgs = (bsConfig, args) => {
+  let local_args = {}
+  local_args['key'] = bsConfig['auth']['access_key'];
+  local_args['localIdentifier'] = bsConfig["connection_settings"]["local_identifier"];
+  local_args['daemon'] = true;
+  local_args['enable-logging-for-api'] = true
+  local_args['source'] = `cypress:${usageReporting.cli_version_and_path(bsConfig).version}`;
+  if(!this.isUndefined(bsConfig["connection_settings"]["local_config_file"])){
+    local_args['config-file'] = path.resolve(bsConfig["connection_settings"]["local_config_file"]);
+  }
+  return local_args;
+};
+
+exports.generateLocalIdentifier = (mode) => {
+  let local_identifier = undefined;
+  if(mode == "always-on"){
+    local_identifier = getmac();
+  } else {
+    local_identifier = uuidv4();
+  }
+  return Buffer.from(local_identifier).toString("base64");
+};
+
+exports.checkLocalIdentifierRunning = (bsConfig, localIdentifier) => {
+  let options = {
+    url: `${config.localTestingUrl}/list?auth_token=${bsConfig.auth.access_key}&state=running`,
+    auth: {
+      user: bsConfig.auth.username,
+      password: bsConfig.auth.access_key,
+    },
+    headers: {
+      'User-Agent': this.getUserAgent(),
+    },
+  };
+  let that = this;
+  return new Promise ( function(resolve, reject) {
+      request.get(options, function (err, resp, body) {
+      if(err){
```

Comment:
Let's indent the code according to their scope.

Reply:
Done